### PR TITLE
sys-libs/libxcrypt: switch to pre-generated autotools tarballs

### PR DESCRIPTION
**Pull request by thesamesam**
There are actually *two* circular dependencies involving Perl:

1) Use self-generated (for now) `make dist` tarballs to avoid a circular
   dependency with libxcrypt->automake->perl->libxcrypt.

   (Thanks juippis and floppym! We noticed this because juippis hit an
   interesting edge case when using binpkgs.)

2) We need to pre-generate the results of a Perl tool called
   during `./configure` in order to avoid unconditionally needing
   Perl.

   We can do this because the input is constant for all of the
   Gentoo build variants - for now.

   (Thanks mattst88! We noticed this while digging into suggestions
   for upstream.)

(Both best explained within the comments of the ebuild.)

Bug: https://bugs.gentoo.org/699422
Reported-by: Joonas Niilola <juippis@gentoo.org>
Reported-by: Mike Gilbert <floppym@gentoo.org>
Reported-by: Matt Turner <mattst88@gentoo.org>
Signed-off-by: Sam James <sam@gentoo.org>

### CODE REVIEW

File: sys-libs/libxcrypt/libxcrypt-4.4.23.ebuild [hunk 1]
```diff
@@ -2,17 +2,27 @@
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
-PYTHON_COMPAT=( python3_{7..10} )
-inherit autotools multibuild python-any-r1 multilib-minimal
+
+PYTHON_COMPAT=( python3_{7..9} )
```
Comment (Arfrever) [inline sys-libs/libxcrypt/libxcrypt-4.4.23.ebuild:6]: Mistakenly removed `python3_10`.

File: sys-libs/libxcrypt/libxcrypt-4.4.23.ebuild [hunk 2]
```diff
@@ -22,13 +32,12 @@ RDEPEND="${DEPEND}"
...
+	# See below comment/notes in src_prepare
+	BDEPEND+=" dev-lang/perl"
+fi
 
-REQUIRED_USE="split-usr? ( system )"
-
-PATCHES=(
-	"${FILESDIR}/libxcrypt-4.4.19-multibuild.patch"
-)
+RESTRICT="!test? ( test )"
```
Comment (Arfrever) [inline sys-libs/libxcrypt/libxcrypt-4.4.23.ebuild:40]: This variable often is set directly after `IUSE` and `REQUIRED_USE`.

**Comment by stefson** [comment]
I had problems with bootstrapping a cross compiler toolchain the other day, because there was USE="-crypt" in the hosts make.conf (for reasons I forgot), and so the cross-glibc was compiled without crypt support. That made stage2-gcc fail when it tried to build its sanitizers. This was before I read the news today, so I'm writing this to just to leave you some feedback.

File: sys-libs/libxcrypt/files/libxcrypt-4.4.20-read-hashes_selected-from-injected-file.patch [hunk 3]
```diff
@@ -0,0 +1,29 @@
+From 6cc88caf1aac5418832af25fe02823a256fe5aa4 Mon Sep 17 00:00:00 2001
```
Comment (thesamesam) [inline sys-libs/libxcrypt/files/libxcrypt-4.4.20-read-hashes_selected-from-injected-file.patch:1]: I just realised: we should hack out the PROG_PERL stuff too.

**Comment by thesamesam** [comment]
@stefson Thanks for bringing this to my attention! Let's discuss on the [bug](https://bugs.gentoo.org/799707) (yours?).

**Pull request closed without merging**